### ISSUE TITLE
Ability to use models built in caret and pdp

### ISSUE THREAD

**Issue by gamrot** [bgreenwell/pdp#62]
Hey,
is there a way to use models built in caret (using train() function) inside pdp? I tried to use the output of train, as well as finalModel from caret's train() output, but nothing works.

**Comment by bgreenwell** [comment]
`pdp` should work just fine with `caret` models. Do you have a small reproducible example?

**Comment by bgreenwell** [comment]
But here is a quick working example:

```r
# Load required packages
library(caret)
library(pdp)
data(boston)

# Quick example
set.seed(101)
mod <- train(
  x = subset(boston, select = -cmedv),
  y = boston$cmedv,
  method = "rf",
  tuneLength = 2,
  trControl = trainControl(method = "cv", number = 5)
)
partial(mod, pred.var = "lstat", plot = FALSE)
#      lstat     yhat
# 1   1.7300 29.88321
# 2   2.4548 29.88252
# 3   3.1796 29.87516
# ...
# 51 37.9700 18.58617
```

**Comment by gamrot** [comment]
Brandon,
thanks for the example. I've found the reason my `caret` model wasn't working --- while training model using caret I've used a `tibble` as an input. Using `as.data.frame()` on data frame used for modelling I was able to solve the problem and produce nice looking partial dependence plot.

I didn't notice but it was already mentioned here: https://github.com/bgreenwell/pdp/issues/59

BTW, in your caret example you use `subset` to select x variable. This works for random forest but not for `xgbLinear` and `xgbTree`. The code below works for all methods:

```
train(cmedv ~ .,
      data = boston,
      method = 'xgbLinear',
      tuneLength = 2,
      trControl = trainControl(method = 'cv',
                               number = 5)
      )
```
